boundary_layer/builders/templates/generator_operator.j2

```diff
@@ -29,23 +29,61 @@ def {{ item_name_builder }}(index, item):
  latter would discard any default task args, expecting them to be filled-in
  by airflow, while in fact airflow would not fill them in at all. #}
 {% set properties = node.resolved_properties.values %}
-for (index, item) in enumerate({{ iterable_builder }}(
+
+{% set all_items = (node.name + '_all_items') | sanitize_operator_name %}
+{{ all_items }} = {{ iterable_builder }}(
 {% for arg in builder_args %}
 {% if arg in properties %}
             {{ arg }} = {{ properties[arg] | format_value }},
 {% endif %}
 {% endfor %}
-        )):
+        )
+
+{% if node.batching.enabled %}
```

Comment:
What if, instead of using a conditional block for this, you could instead:
- use a default batch size of 1
- create the `{{item_name}}` variable inside the builder function only if the batch size is 1
- have the batch-name builder default to just choosing the item_name if the batch size is 1?

I think something along these lines might simplify a lot of the code, because many lines and branches have to go into dealing with differences between `item_name` and `batch_name`.  I guess `batch_name` would become standard and `item_name` would only be a special case.

Reply:
Alright yeah you convinced me 😄 

I think I still do prefer the interface described below in the comment about `BatchingSchema`, where if `batch_size` is missing or `None` then we implicitly interpret that to be equivalent to disabling batching, so that we don't need to fill in both the `enabled` field and the `batch_size` field.  thoughts on that?

Reply:
I like it.  Makes usage a little bit easier.  👍